src/main/java/eggventory/Eggventory.java

```diff
@@ -65,13 +73,17 @@ private static void userInteraction() {
         try {
 
             String userInput = ui.read();
-
+            //TODO: Check whether SLAP is violated
             Command command = parser.parse(userInput);
             if (command.getType().equals(CommandType.BYE)) {
-                ((ByeCommand) command).executeSaveMoreLists(stockList, ui, storage, loanList, personList, templateList);
+                ((ByeCommand) command).executeSaveMoreLists(stateInterface.getStockList(), ui, storage,
+                        stateInterface.getLoanList(), stateInterface.getPersonList(),
+                        stateInterface.getTemplateList());
             }
-            command.execute(stockList, ui, storage);
-            storage.save(stockList, loanList, personList, templateList);
```

Comment:
i would prefer if you kept this line in unless it affects anything that you're doing